test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -1554,14 +1692,212 @@ def test_cgroup_assign_resources_mem_only_vnode(self):
         j3 = Job(TEST_USER, attrs=a)
         j3.create_script('date')
         jid3 = self.server.submit(j3)
-
         a = {'job_state': 'Q',
              'comment':
              (MATCH_RE,
               '.*Can Never Run: Insufficient amount of resource: mem.*')}
         self.server.expect(JOB, a, attrop=PTL_AND, id=jid3, offset=10,
                            interval=1, max_attempts=30)
 
+    @timeout(300)
+    def test_cgroup_cpuset_exclude_cpu(self):
+        """
+        Confirm that exclude_cpus reduces resources_available.ncpus
+        """
+        # Fetch the unmodified value of resources_available.ncpus
+        self.load_config(self.cfg5 % ('false', '', 'false', 'false',
+                                      'false', self.swapctl))
+        self.server.expect(NODE, {'state': 'free'},
+                           self.hostA.shortname, interval=3, offset=10)
+        result = self.server.status(NODE, 'resources_available.ncpus')
+        orig_ncpus = 0
+        for entry in result:
+            if entry['id'] == self.hostA.shortname:
+                orig_ncpus = int(entry['resources_available.ncpus'])
+                break
+        self.assertTrue(orig_ncpus > 0)
+        self.logger.info('Original value of ncpus: %d' % orig_ncpus)
+        if orig_ncpus < 2:
+            self.skipTest('Node must have at least two CPUs')
+        # Now exclude CPU zero
+        self.load_config(self.cfg5 % ('false', '0', 'false', 'false',
+                                      'false', self.swapctl))
+        self.server.expect(NODE, {'state': 'free'},
+                           self.hostA.shortname, interval=3, offset=10)
+        result = self.server.status(NODE, 'resources_available.ncpus')
+        new_ncpus = 0
+        for entry in result:
+            if entry['id'] == self.hostA.shortname:
+                new_ncpus = int(entry['resources_available.ncpus'])
+                break
+        self.assertTrue(new_ncpus > 0)
+        self.logger.info('New value with one CPU excluded: %d' % new_ncpus)
+        self.assertTrue((new_ncpus + 1) == orig_ncpus)
+        # Repeat the process with vnode_per_numa_node set to true
+        vnode = '%s[0]' % self.hostA.shortname
+        self.load_config(self.cfg5 % ('true', '', 'false', 'false',
+                                      'false', self.swapctl))
+        self.server.expect(NODE, {'state': 'free'},
+                           vnode, interval=3, offset=10)
+        result = self.server.status(NODE, 'resources_available.ncpus')
+        orig_ncpus = 0
+        for entry in result:
+            if entry['id'] == vnode:
+                orig_ncpus = int(entry['resources_available.ncpus'])
+                break
```

Comment:
same as above, use id=vnode and skip the for loop

Reply:
Done.